projects/cashmere/src/lib/tabs/tab-set/tab-set.component.ts

```diff
@@ -499,29 +506,48 @@ export class TabSetComponent implements AfterContentInit {
     }
 
     private defaultToFirstRoute() {
-        const foundRoute = this._tabs
-            .map(tab => tab.routerLink)
-            .map(routerLink => this.mapRouterLinkToString(routerLink))
-            .find(routerLink => {
-                const currentRoute = this.router.url;
-                return currentRoute === routerLink || currentRoute.indexOf(`${routerLink}/`) > -1;
-            });
+        const tabArray = this._tabs.toArray();
+        let routeTab = -1;
+
+        // Determine if there is a routerLink in our tabset that matches the current route
+        for ( let i=0; i < tabArray.length; i++ ) {
+            const routerLink = this.mapRouterLinkToString(tabArray[i].routerLink);
+            const currentRoute = this.router.url.split("?")[0];
+            if( currentRoute === routerLink ) {
```

Comment:
We should consider doing a case-insensitive comparison here.

Reply:
Good call - probably just `toLowerCase()` on both sides?